core.js

```diff
@@ -611,17 +611,24 @@ class FileTypeParser {
 		}
 
 		if (this.checkString('%PDF')) {
-			await tokenizer.ignore(1350);
-			const maxBufferSize = 10 * 1024 * 1024;
-			const buffer = Buffer.alloc(Math.min(maxBufferSize, tokenizer.fileInfo.size));
-			await tokenizer.readBuffer(buffer, {mayBeLess: true});
+			try {
+				await tokenizer.ignore(1350);
```

Comment:
I think only this line should be between the `try` and `catch`.

Maybe a more fundamental question is, why it is ignoring the first 1350 bytes if this does not represent a  portion of a PDF which is always present.

The `readBuffer` with the `mayBeLess: true` should not raise an EOF.

Reply:
It shouldn't, but unfortunately it does

```
  fixture-minimal.pdf 295 .fromStream() method - same fileType

  test.js:300

   299:   const file = path.join(__dirname, 'fixture', filename);
   300:   const fileType = await fileTypeFromStream(fs.createReadStream(file)…
   301:

  Rejected promise returned by test. Reason:

  Error (EndOfStreamError) {
    message: 'End-Of-Stream',
  }

  › StreamReader.read (file://node_modules/peek-readable/lib/StreamReader.js:49:19)
  › ReadStreamTokenizer.readBuffer (file://node_modules/strtok3/lib/ReadStreamTokenizer.js:35:51)
  › FileTypeParser.parse (file://core.js:625:20)
  › async fileTypeFromStream (file://core.js:16:10)
  › async testFileFromStream (file://test.js:300:19)
```

failure from my test case when i only try the ignore

Reply:
I'm just going to early return pdf on the catch instead.

Reply:
If continue to read of the EOF is already caught, you can expect problems.

Sorry, I think you initial code was actually better. Declaring the the PDF return type twice is very nice neither. If you revert to the previous point I approve.